packages/algoliasearch-helper/index.d.ts

```diff
@@ -1553,6 +1554,20 @@ declare namespace algoliasearchHelper {
       params: RecommendParametersWithId<LookingSimilarQuery>
     ): RecommendParameters;
   }
+
+  type RecommendResponse<TObject> =
+    RecommendQueriesResponse<TObject>['results'];
+
+  type RecommendResultItem<TObject = any> = RecommendResponse<TObject>[0];
+
+  export class RecommendResults<T = any> {
+    constructor(state: RecommendParameters, results: RecommendResponse<T>);
+
+    _state: RecommendParameters;
+    _rawResults: RecommendResponse<T>;
+
+    [index: number]: RecommendResultItem<T>;
```

Comment:
Can't have `[index: string]` as there would be a problem with `_state` and `_rawResults`

Reply:
Strings would have helped us prefix the id to prevent conflicts between different Recommend widget types. How would we solve this with simple numbers? Maybe we could have a global counter somewhere, and/or have the `$$id` be filled in by the index widget at `addWidgets()` time?

Reply:
As discussed we're keeping numerical internal ids for the time being. I'll resolve this conversation.